### ISSUE TITLE
`class` property not applied to component when Ivy is enabled

### ISSUE THREAD

**Issue by madoar** [angular/angular#36400]
# 🐞 bug report

### Affected Package
I'm not sure

### Is this a regression?
Yes, because Ivy was not present in previous versions

### Description
I'm the main author of an Angular library named `angular-archwizard`.
A few users of the library stumbled over the issue where the newest version of wizard, which has been built using Angular 9, does not seem to work when used in another project using Angular 9 and Ivy deactivated. 

### Ivy off
![grafik](https://user-images.githubusercontent.com/18488086/78283547-38c31880-751e-11ea-8efa-453afb4bbd21.png)

### Ivy on
![grafik](https://user-images.githubusercontent.com/18488086/78283613-53958d00-751e-11ea-9640-515ba62d0d0f.png)

## 🔬 Minimal Reproduction
https://stackblitz.com/edit/angular9-archwizard-css?file=tsconfig.json

Flip `enableIvy` to `false` to see that the css classes are not applied anymore

## 🔥 Exception or Error
```
WARNING: sanitizing unsafe style value [object Object] (see http://g.co/ng/security#xss).
```

In addition it seems like the `class` property of the `aw-wizard-navigation-bar` is not applied correctly. Instead of `horizontal small` it contains `[object Object]` (see also https://github.com/madoar/angular-archwizard/issues/278#issuecomment-607490550).

Maybe the warning and the `class` property issue are connected?

## 🌍  Your Environment
Link to the repository hosting `angular-archwizard`: https://github.com/madoar/angular-archwizard

**Anything else relevant?**
I'm not sure whether this is actually a bug in Angular or maybe a misconfiguration of `angular-archwizard`. Independent of this I think it is strange that also the tests of `angular-archwizard` only seem to work with Ivy enabled. If you disable it by adding 

```json
"angularCompilerOptions": {
  "enableIvy": false
}
```

to for example [`tsconfig`](https://github.com/madoar/angular-archwizard/blob/develop/tsconfig.json) a number of tests of `angular-archwizard` are failing as can be seen [here](https://github.com/madoar/angular-archwizard/runs/555130754).

**Comment by madoar** [comment]
@JoostK thank you for looking into this!

I thought starting with Angular 9 `class` was supported for Ivy and the old ViewEngine, but it seems like I was wrong with that assumption. This leads me to two followup questions:

-  is there a way to stay with the in my opinion more intuitive `class` parameter or is the only solution to switch back to `ngClass` if I want to support both the ViewEngine and Ivy renderer?
-  is there an easy way to execute my tests using both the ViewEngine and the Ivy renderer? Currently it seems to me that I need to decide which renderer I want to use beforehand in my config file(s).

**Comment by JoostK** [comment]
> * is there a way to stay with the in my opinion more intuitive `class` parameter or is the only solution to switch back to `ngClass` if I want to support both the ViewEngine and Ivy renderer?

Yes, I'm afraid you'd have to switch back to using `ngClass`. The downside of running a library is that you need to conform to the requirements if the earliest supported version. Just to note here, even compiling with Angular 9.0 using the ViewEngine compiler will work in Angular 8 and earlier.

> * is there an easy way to execute my tests using both the ViewEngine and the Ivy renderer? Currently it seems to me that I need to decide which renderer I want to use beforehand in my config file(s).

You would indeed need different configuration files. I see that `ng test` has the `--tsConfig` option, so you could create a `tsconfig.spec.ve.json` next to `tsconfig.spec.json` with something along the line of:

```json
{
  "extends": "./tsconfig.spec.json",
  "angularCompilerOptions": {
    "enableIvy": false,
  }
}
```

and then run the tests twice, where you point to this custom tsconfig in one of the runs. I haven't tested this so I'm not certain it would work out.

**Comment by madoar** [comment]
> Just to note here, even compiling with Angular 9.0 using the ViewEngine compiler will work in Angular 8 and earlier

Sadly this does not seem to be the case for `angular-archwizard` because of a [breaking change](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#class-field-mitigations) in the required `typescript` version. More specific `angular-archwizard` makes heavy use of `get` and `set` accessor methods. Starting with `typescript` `3.7` these accessor methods are emitted differently in the `.d.ts` files. This results in [error messages](https://github.com/madoar/angular-archwizard/issues/279) if you try to use `angular-archwizard` built with Angular 9 in a web application built with Angular 8

**Comment by JoostK** [comment]
@madoar Oh no, the most crucial word in that sentence is missing! I meant to say it will **not** work in Angular 8. TypeScript differences may indeed contribute to this, but in this particular case the v9 compiled output for tree-shakeable providers has changed so those would be invalid in an Angular 8 app.
